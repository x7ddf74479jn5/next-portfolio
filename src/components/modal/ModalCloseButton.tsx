import styles from "src/styles/components/modal/ModalCloseButton.module.scss";
import type { ButtonType } from "src/types/utils";

type ModalCloseButtonProps = ButtonType;

const ModalCloseButton: React.VFC<ModalCloseButtonProps> = (props) => {
  return (
    <button className={styles.closeButton} {...props}>
      Ã—
    </button>
  );
};

export default ModalCloseButton;
// TODO: test
