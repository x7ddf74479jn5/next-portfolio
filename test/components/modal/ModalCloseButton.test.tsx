import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ModalCloseButton from "src/components/modal/ModalCloseButton";

describe("ModalCloseButton", () => {
  let renderResult: RenderResult;
  let buttonElement: HTMLElement;
  const mockHandleClick = jest.fn();

  beforeEach(() => {
    renderResult = render(<ModalCloseButton onClick={mockHandleClick} />);
    buttonElement = renderResult.getByRole("button");
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(buttonElement).toHaveTextContent("Ã—");
    expect(buttonElement).toBeEnabled();
  });

  it("user interaction", () => {
    userEvent.click(buttonElement);
    expect(mockHandleClick).toHaveBeenCalled();
  });
});
