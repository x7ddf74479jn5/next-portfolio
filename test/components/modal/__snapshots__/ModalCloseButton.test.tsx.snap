// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCloseButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="closeButton"
  >
    Ã—
  </button>
</DocumentFragment>
`;
